### PR TITLE
Stabilize method_disconnect end-to-end tests

### PR DESCRIPTION
**Pull request by pierreca**
- add logs
- add timeouts
- manually open device client

I've had pretty good success with this (passed 90 times in a row, with logging enabled and disabled)

### CODE REVIEW

File: e2etests/test/method_disconnect.js [hunk 1]
```diff
@@ -171,30 +171,27 @@ protocolAndTermination.forEach( function (testConfiguration) {
...
-                  sendMethodCallback(err);
-                });
-      }, 1000);
+          provisionedDevice.deviceId,
+          methodParams,
+          function(err, result) {
+            if(!err) {
+              debug('got method results');
+              debug(JSON.stringify(result, null, 2));
+              assert.strictEqual(result.status, methodResult);
```
Comment (anthonyvercolano) [inline e2etests/test/method_disconnect.js]: >result.status, methodResult [](start = 33, length = 27)

I would like to add a 'policy' to e2e tests as much as possible.  I want to see the device id printed out with every failure.  This way we can examine the logs and have the device id as one of the search parameters.

File: e2etests/test/method_disconnect.js [hunk 2]
```diff
@@ -211,20 +208,35 @@ protocolAndTermination.forEach( function (testConfiguration) {
...
+      deviceClient.open(function (err) {
+        if (err) {
+          debug('error connecting the device client: ' + err.toString());
           testCallback(err);
+        } else {
+          debug('setting up method handler for ' + methodName1);
+          setMethodHandler(methodName1, firstPayload);
+          debug('waiting 2 seconds before invoking ' + methodName1);
+          setTimeout(function () {
+            sendMethodCall(serviceClient, methodName1, firstPayload, function(err) {
```
Comment (anthonyvercolano) [inline e2etests/test/method_disconnect.js]: >sendMethodCall(serviceClient, methodName1, firstPayload, function(err) { [](start = 12, length = 72)

You might want to use the rendezvous helper on these.   I've certainly seen the response from the injection come in quite late.  You might think of this as having three paths here.  The injection path, the client method handler path, and the service sending the method path.  Just a thought.

File: e2etests/test/method_disconnect.js (hunk 1 above)
Comment (pierreca) [reply e2etests/test/method_disconnect.js]: that's a good idea. let's start doing that.

File: e2etests/test/method_disconnect.js (hunk 2 above)
Comment (pierreca) [reply e2etests/test/method_disconnect.js]: I'll add that if we see failures.

**Pull request merged**